src/app/modules/item/pages/item-log-view/item-log-view.component.ts

```diff
@@ -26,12 +27,8 @@ export class ItemLogViewComponent implements OnChanges {
   @Input() itemData?: ItemData;
   @Input() isWatchingGroup = false;
 
-  private logColumns?: Column[];
-
-  private readonly id$ = new ReplaySubject<string>(1);
-  readonly state$ = this.id$.pipe(
-    distinct(),
-    switchMap(id => this.getData$(id)),
+  private readonly item$ = new ReplaySubject<Item>(1);
+  readonly state$ = this.getData$().pipe(
```

Comment:
I work for more than 5years with Rx (not just in JS) and actually if you do not put the `this.item$.pipe...` here, you lose most of the power of Rx (it is more a functional programming thing than Rx)... actually mainly you loose its readability (and Rx can quickly be an unreadable mess in some cases).
We need to see at first sight that `state$` is a steam directly based on `item$`. 
Please move the `getData$` code here. Anyway that's also for code consistency with the rest of the project.

Reply:
Fixed